### PR TITLE
Add purchase order store trait and structs

### PR DESCRIPTION
**Pull request by dnewh**
This adds the PurchaseOrderStore trait and associated structs. This includes the native models of PurchaseOrder and PurchaseOrderAlternateID as well as the PurchaseOrderStoreError type.

### CODE REVIEW

File: sdk/src/migrations/diesel/postgres/migrations/2021-03-24-130200_purchase_order_tables/up.sql [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+-- -----------------------------------------------------------------------------
+
+CREATE TABLE purchase_order (
+    id BIGSERIAL PRIMARY KEY,
+    uid TEXT NOT NULL,
```
Comment (shannynalayna) [inline sdk/src/migrations/diesel/postgres/migrations/2021-03-24-130200_purchase_order_tables/up.sql:18]: Needs to be changed to `uuid`

File: sdk/src/purchase_order/store/mod.rs [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+/// Represents a list of Grid Purchase Orders
+#[derive(Clone, Debug, Serialize, PartialEq)]
+pub struct PurchaseOrderList {
+    pub purchase_orders: Vec<PurchaseOrder>,
+}
+
+/// Represents a Grid Purchase Order
+#[derive(Clone, Debug, Serialize, PartialEq)]
+pub struct PurchaseOrder {
+    pub uid: String,
```
Comment (shannynalayna) [inline sdk/src/purchase_order/store/mod.rs:32]: `uid` -> `uuid`

File: sdk/src/purchase_order/store/mod.rs [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    pub end_commit_num: i64,
+    pub service_id: Option<String>,
+}
+
+/// Represents a Grid Purchase Order Version
+#[derive(Clone, Debug, Serialize, PartialEq)]
+pub struct PurchaseOrderVersion {
+    pub version_id: String,
+    pub is_draft: bool,
+    pub current_revision_number: i64,
```
Comment (shannynalayna) [inline sdk/src/purchase_order/store/mod.rs:49]: `revision_number` -> `revision_id`, same goes for the `_Revision` struct as well

File: sdk/src/migrations/diesel/postgres/migrations/2021-03-24-130200_purchase_order_tables/up.sql (hunk 1 above)
Comment (dnewh) [reply sdk/src/migrations/diesel/postgres/migrations/2021-03-24-130200_purchase_order_tables/up.sql]: updated

Comment (dnewh) [reply sdk/src/migrations/diesel/postgres/migrations/2021-03-24-130200_purchase_order_tables/up.sql]: updated

File: sdk/src/purchase_order/store/mod.rs (hunk 2 above)
Comment (dnewh) [reply sdk/src/purchase_order/store/mod.rs]: updated

File: sdk/src/purchase_order/store/mod.rs (hunk 3 above)
Comment (dnewh) [reply sdk/src/purchase_order/store/mod.rs]: updated

**Review by vaporos — APPROVED** [review]

**Review by shannynalayna — APPROVED** [review]

**Pull request merged**